sdks/csharp/queries.cs

```diff
@@ -11,52 +11,49 @@
 try
 {
     await foreach(BasicDigitalTwin twin in result)
+    {
+        // You can include your own logic to print the result
+        // The logic below prints the twin's ID and contents
+        Console.WriteLine($"Twin ID: {twin.Id} \nTwin data");
+        foreach (KeyValuePair<string, object> kvp in twin.Contents)
         {
-            // You can include your own logic to print the result
-            // The logic below prints the twin's ID and contents
-            Console.WriteLine($"Twin ID: {twin.Id} \nTwin data");
-            IDictionary<string, object> contents = twin.Contents;
-            foreach (KeyValuePair<string, object> kvp in contents)
-            {
-                Console.WriteLine($"{kvp.Key}  {kvp.Value}");
-            }
+            Console.WriteLine($"{kvp.Key}  {kvp.Value}");
         }
+    }
 }
-catch (RequestFailedException e)
+catch (RequestFailedException ex)
 {
-    Console.WriteLine($"Error {e.Status}: {e.Message}");
+    Console.WriteLine($"Error {ex.Status}, {ex.ErrorCode}, {ex.Message}");
     throw;
 }
 // </FullQuerySample>
 
 // ------------------ Paging pattern for query API ---------------------
 // <PagedQuery>
-string query = "SELECT * FROM digitaltwins";
-string conToken = null; // continuation token from the query
-int page = 0;
 try
 {
     // Repeat the query while there are pages
-    do
+    string query = "SELECT * FROM digitaltwins";
+    AsyncPageable<BasicDigitalTwin> queryResult = await client.QueryAsync(query);
+    int page = 0;
+    await foreach (Page<BasicDigitalTwin> page in queryResult.AsPages())
     {
-        QuerySpecification spec = new QuerySpecification(query, conToken);
-        QueryResult qr = await client.Query.QueryTwinsAsync(spec);
-        page++;
-        Console.WriteLine($"== Query results page {page}:");
-        if (qr.Items != null)
+        Console.WriteLine($"== Query results page {++page}:");
+
+        // Extract the query-charge header from the page
+        if (QueryChargeHelper.TryGetQueryCharge(page, out float queryCharge))
```

Comment:
I added in how to get the query charge. LMK if you'd rather this not be in there. It is the main reason I can think of to get pages.

Reply:
Seems fine to me. We use this example [here](https://docs.microsoft.com/azure/digital-twins/how-to-create-custom-sdks#paging) just to show a paging pattern, so it doesn't mess with the context to include this. I can then add a little extra corresponding info explaining that query charge is the main reason you'd do this.

Reply:
I think this is all used for web documentation so I'm not sure they want that much context in the code. @baanders and @alexkarcher-msft ?

Reply:
All of the code snippets are used in the online [docs](https://docs.microsoft.com/azure/digital-twins/) in the context of a larger document, so I was just noting that this addition still makes sufficient sense with the rest of the text that goes along with this snippet and what we're using it to illustrate.